packages/server/src/threads/automation.ts

```diff
@@ -192,6 +203,57 @@ class Orchestrator {
     }
   }
 
+  handleLooping(
+    loopSteps: LoopStep[] | undefined,
+    iterationCount: number,
+    loopStep: LoopStep | undefined,
+    loopStepNumber: number,
+    index: number,
+    originalStepInput: any,
+    step: any
```

Comment:
I feel like in general lot of the `any` is coming from the lack of step/input typing - I think if we fixed these then it would be a lot easier to fill out the typing on the `loopUtils` - would be worthwhile improving the typing as part of this refactor as it just provides a good bit more safety.